src/tbb/parallel_pipeline.cpp

```diff
@@ -243,7 +243,8 @@ class input_buffer {
         return end_of_input_tls.get() != 0;
     }
     void set_my_tls_end_of_input() {
-        end_of_input_tls.set(1);
+	// Use an arbitrary but valid pointer as value.
+        end_of_input_tls.set(array_size);
```

Comment:
I am not sure that I understand the waring and approach how it is fixed (`array_size` is also integral type similar to `1`).

Reply:
Thank you for the explanation.

It seems to be specific to `pthread_setspecific`. Can we try to encapsulate the workaround into `basic_tls`? Will it work if we rework implementation near [`tbb/tls.h::44`](https://github.com/oneapi-src/oneTBB/blob/master/src/tbb/tls.h#L44)?:
```diff 
-void set( T value ) { pthread_setspecific(my_key, (void*)value); }
+// Promote `value` into pointer-size integral to prevent GCC 11+ warning.
+void set( T value ) { pthread_setspecific(my_key, (void*)(std::uintptr_t)value; }
```

Reply:
Can you please rebase on the master branch?

Reply:
I don't this it will work. problem is the value being passed not the type.

Reply:
For some unclear reason I cannot reproduce the issue on godbolt: https://godbolt.org/z/rYseqo367

I am slightly confused that we need some changes on user side of `tls`. It is implementation detail that `tls` uses `pthread_setspecific` (on other systems it uses something else). Moreover, the `my_tls_end_of_input` function (line 243) expects some non-zero value, maybe `array_size` is not zero, but it seems strange to pass something unknown that affect program flow. Do not we have some other workarounds inside the `tls::set` function to resolve the warning, e.g. suppress with compiler specific pragmas if there is no better solution?

Reply:
you need glibc/master ( soon to be 2.34 ) release

Reply:
The concern [is addressed](https://github.com/oneapi-src/oneTBB/pull/519/files#diff-2595b33d006f9bcc276fdb214d03b3c8f7cd7ccde63358b534ad69bccb20d2b8R242-R247) in #519